cumulusci/tasks/apex/custom_settings.py

```diff
@@ -0,0 +1,126 @@
+""" a task for waiting on a specific custom settings value """
+
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.exceptions import SalesforceException
+from simple_salesforce import SalesforceMalformedRequest
+from cumulusci.core.utils import process_bool_arg
+
+
+class CustomSettingValueWait(BaseSalesforceApiTask):
+    """ CustomSettingValueWait polls an org until the specific value exists in a custom settings field """
+
+    name = "CustomSettingValueWait"
+
+    task_options = {
+        "object": {
+            "description": "Name of the Hierarchical Custom Settings object to query. Can include the %%%NAMESPACE%%% token. ",
+            "required": True,
+        },
+        "field": {
+            "description": "Name of the field on the Custom Settings to query. Can include the %%%NAMESPACE%%% token. ",
+            "required": True,
+        },
+        "value": {
+            "description": "Value of the field to wait for (String, Integer or Boolean). ",
+            "required": True,
+        },
+        "managed": {
+            "description": (
+                "If True, will insert the project's namespace prefix.  "
+                "Defaults to False or no namespace."
+            ),
+            "required": False,
+        },
+        "namespaced": {
+            "description": (
+                "If True, the %%%NAMESPACE%%% token "
+                "will get replaced with the namespace prefix for the object and field."
+                "Defaults to False."
+            ),
+            "required": False,
+        },
+        "poll_interval": {
+            "description": (
+                "Seconds to wait before polling for batch job completion. "
+                "Defaults to 10 seconds."
+            ),
+        },
+    }
+
+    def _run_task(self):
+        self.poll_interval_s = int(self.options.get("poll_interval", 10))
+
+        # Retrieve polling object/field/value options
+        self.object_name = self.options["object"]
+        self.field_name = self.options["field"]
+        self.check_value = self.options["value"]
+
+        # Process namespace tokens
+        self._apply_namespace()
+
+        # will block until poll_complete
+        self._poll()
+
+        self.logger.info("Value Matched.")
+
+        return True
+
+    def _poll_action(self):
+        try:
+            query_results = self.sf.query(self._object_query)
+        except:
+            print(
+                "Only Hierarchical Custom Settings objects can be used with this task"
+            )
+            raise
```

Comment:
Oh, and the import: `from cumulusci.core.exceptions import TaskOptionsError`

Reply:
Thanks. Is there a way to get more about the error since it could be a bad object or field name too? At one point I tried to parse the error response, but didn't have any luck with it.

Reply:
Thanks for the tips on this. I added some better error parsing to provide a more accurate error.